File: frontend/src/lib/types/film.ts
export interface Film {
  title: string;
  tmdbId: string;
  country?: string;
  year?: string;
  posterUrl?: string;
  overview?: string;
  genres?: string[];
  runtime?: number;
  trailerUrl?: string;
  isInRadarr?: boolean;
  scrapedDate?: Date;
}

File: backend/CinemaList.Common/Models/Film.cs
using System;
using System.Diagnostics.CodeAnalysis;
using MongoDB.Bson.Serialization.Attributes;

namespace CinemaList.Common.Models;

[BsonIgnoreExtraElements]
public class Film
{
    [BsonId]
    [BsonRepresentation(MongoDB.Bson.BsonType.ObjectId)]
    [AllowNull]
    public string Id { get; set; } = null!;
    public required string Title { get; set; } = string.Empty;
    
    public required string TmdbId { get; set; } = string.Empty;
    
    public bool IsInRadarr { get; set; } = false;

    public string? Country { get; set; }
    public string? Year { get; set; }
    
    public string? PosterUrl { get; init; }
    
    public required DateTime ScrapedDate { get; init; }
}